frontend-minesweeper/src/App.js

```diff
@@ -0,0 +1,168 @@
+        }));
+        setTileData(updatedTiles);
+      }
+    }
+  }, [apiResponse]);
+
+  const createNewGame = () => {
+    if (intervalRef.current) clearInterval(intervalRef.current);
+    intervalRef.current = null;
+    setApiResponse(null);
+    setGameWon(false);
+    setGameOver(false);
+    setShowPopup(false);
+    setTileData([]);
+    setTime(0);
+    setTimerOn(false);
+    const data = { level, userName: "sigmaronny" };
+
+    axios.post(`${API_HOST}/games/`, data)
+      .then(response => {
+        setApiResponse(response.data);
+        setGameId(response.data.gameId);
+        if (response.data.board && response.data.board.tiles) {
+          setTileData(response.data.board.tiles);
+        }
+        setTimerOn(true);
+      })
+      .catch(error => console.error("Feil ved oppretting av spill:", error));
+  };
+
+  const handleTileAction = (row, column, move) => {
+    if (gameWon || gameOver) { return; }
+    const flaggedTiles = tileData.filter(tile => tile.isFlagged).length;
+    if (move === 1 && flaggedTiles >= gameSettings[level].mines) {
+      return;
+    }
+    const MoveData = { row, column, move };
+    axios.put(`${API_HOST}/games/${gameId}`, MoveData)
+      .then(response => {
+        setApiResponse(response.data);
+        if (response.data.board && response.data.board.tiles) {
+          setTileData(response.data.board.tiles);
+        }
+        checkVictory();
+      })
+      .catch(error => console.error("Feil ved oppdatering av flis:", error));
+  };
+
+  const checkVictory = () => {
+    if (apiResponse?.gameWon) {
+      setGameWon(true);
+      setTimerOn(false);
+    }
+  };
+
+  const getNumberColor = (number) => {
+    const colors = ["", "#0000FF", "#008000", "#FF0000", "#800080", "#800000", "#40E0D0", "#000000", "#808080"];
+    return colors[number];
+  };
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <div className='game'>
+          <h1>Altinn II<br />Bomberydder</h1>
+          <select className='level-select' value={level} onChange={e => setLevel(Number(e.target.value))}>
+            <option value={0}>Nybegynner</option>
+            <option value={1}>Mellomnivå</option>
+            <option value={2}>Ekspert</option>
+          </select>
+          <div className='game-info'>
+            <div className='timer'>{String(time).padStart(3, '0')}</div>
+            <button className='reset' onClick={createNewGame}>{gameWon ? <img src={winner} height='30px' alt='' /> : gameOver ? '😵' : '🙂'}</button>
+            <div className='mines'>{String(gameSettings[level].mines - tileData.filter(t => t.isFlagged).length).padStart(3, '0')}</div>
+          </div>
+
+          <div className='grid' style={{
+            gridTemplateColumns: `repeat(${gameSettings[level].columns}, 30px)`,
+            gridTemplateRows: `repeat(${gameSettings[level].rows}, 30px)`
+          }}>
+            {tileData.map((tile, index) => (
```

Comment:
Veldig mye som skjer i denne map'en. Mye logikk for å bestemme hvordan hver tile skal se ut, bør flyttes ut som egen komponent.